frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx

```diff
@@ -0,0 +1,117 @@
+import * as _ from 'lodash';
+
+import {
+  CreateVmWizard,
+  getResource,
+  TEMPLATE_TYPE_LABEL,
+  TEMPLATE_TYPE_VM,
+  TEMPLATE_TYPE_BASE,
+} from 'kubevirt-web-ui-components';
+
+import {
+  NamespaceModel,
+  TemplateModel,
+  PersistentVolumeClaimModel,
+  StorageClassModel,
+} from '@console/internal/models';
+import { getName, createModalResourceLauncher } from '@console/shared';
+import { units, Firehose } from '@console/internal/components/utils';
+import {
+  k8sCreate,
+  k8sGet,
+  k8sPatch,
+  k8sKill,
+  K8sResourceKind,
+  K8sKind,
+} from '@console/internal/module/k8s';
+import { ModalComponentProps } from '../../../../../public/components/factory/modal';
+import {
+  VirtualMachineModel,
+  NetworkAttachmentDefinitionModel,
+  DataVolumeModel,
+} from '../../models';
+
+export const openCreateVmWizard = (activeNamespace, createTemplate = false) => {
+  const resources = [
+    getResource(NamespaceModel, { prop: 'namespaces', optional: true }),
+    getResource(VirtualMachineModel, { prop: 'virtualMachines' }),
+    getResource(TemplateModel, {
+      namespace: activeNamespace,
+      prop: 'userTemplates',
+      matchLabels: { [TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_VM },
+    }),
+    getResource(TemplateModel, {
+      namespace: 'openshift',
+      prop: 'commonTemplates',
+      matchLabels: { [TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_BASE },
+    }),
+    getResource(NetworkAttachmentDefinitionModel, {
+      namespace: activeNamespace,
+      prop: 'networkConfigs',
+    }),
+    getResource(StorageClassModel, { prop: 'storageClasses' }),
+    getResource(PersistentVolumeClaimModel, {
+      namespace: activeNamespace,
+      prop: 'persistentVolumeClaims',
+    }),
+    getResource(DataVolumeModel, { namespace: activeNamespace, prop: 'dataVolumes' }),
+  ];
+
+  const resourcesToProps = (props) => {
+    const { namespaces, userTemplates, commonTemplates } = props;
+    const flatten = (resourceName) => ({ [resourceName]: _.get(props[resourceName], 'data') });
+
+    let selectedNamespace;
+    if (activeNamespace) {
+      selectedNamespace = _.get(namespaces, 'data', []).find(
+        (namespace) => getName(namespace) === activeNamespace,
+      );
+    }
+
+    let templates;
+    if (userTemplates && commonTemplates) {
+      templates = _.get(userTemplates, 'data', []).concat(_.get(commonTemplates, 'data', []));
+    }
+
+    return {
+      selectedNamespace,
+      templates,
+
+      ...flatten('namespaces'),
+      ...flatten('virtualMachines'),
+      ...flatten('userTemplates'),
+      ...flatten('commonTemplates'),
+      ...flatten('networkConfigs'),
+      ...flatten('storageClasses'),
+      ...flatten('persistentVolumeClaims'),
+      ...flatten('dataVolumes'),
+    };
+  };
+
+  type CreatVMWizardResourceProps = {
```

Comment:
Can we move this outside of the function?

Reply:
Sure :exclamation:  but it's just for play, #1822 may change and make all this obsolete :unicorn:

Reply:
ok then

Reply:
> ok then

I did move it out :-)